crates/pixi_config/src/gateway.rs

```diff
@@ -19,3 +22,19 @@ pub fn from_pixi_config(config: &Config) -> ChannelConfig {
         per_channel: Default::default(),
     }
 }
+
+/// Constructs a [`Gateway`] from a [`ClientWithMiddleware`] and a [`Config`]
+pub fn new_gateway(client: ClientWithMiddleware, config: Config) -> Gateway {
```

Comment:
I think you can take the config by reference and use the `from_pixi_config` function in the `with_channel_config` function. No need to move ownership.